### PR TITLE
Add references and fix direct product

### PR DESCRIPTION
**Pull request by Akshobhya1234**
Clearly differentiate product, direct product, and direct sum with definitions in the literature. 
Give reference to the current issue and state what is out of scope of the thesis.

### CODE REVIEW

**Review by JacquesCarette — CHANGES_REQUESTED** [review]
So the section is still titled 'Direct Product' and it refers to another chapter which still talks about 'direct product'. You should make the thesis self-consistent.

The definition of product from Sanella is the 'usual' one that most expect, i.e. Cartesian product.

**Comment by Akshobhya1234** [comment]
> So the section is still titled 'Direct Product' and it refers to another chapter which still talks about 'direct product'. You should make the thesis self-consistent.
> 
> The definition of product from Sanella is the 'usual' one that most expect, i.e. Cartesian product.

I call it 'Direct Product' since they are called as 'direct product' in the standard library. I give the definition of 'product' from Sanella and say that when we talk about 'direct product' we are dealing with product. The definition in Chapter 2 is the correct definition of 'Direct Product' from Sanella. 

The term 'Cartesian product' is usually used when dealing with sets and the term 'Product' is used for algebra. I also clarify this in the same section.

The same goes for 'Morphism'. I call the section 'morphism' since they are available under that module in the standard library.

**Comment by JacquesCarette** [comment]
You shouldn't take the standard library as "conveyor of truth", and indeed, you should criticize it for choosing an odd name here and there. You should take a standard textbook (like Sanella) as establishing standard vocabulary instead.

If you insist on keeping 'direct product' in there, then you need to say (early and often) that this choice is to be consistent with the naming in the standard library. And you should still criticize that non-standard choice!

'Cartesian product' occurs all over category theory, and is way way more general than in just set theory.

**Comment by Akshobhya1234** [comment]
I added definitions of product, direct product, and direct sum in the universal algebra chapter. I changed the section to Product Algebra instead of Direct Product.

File: Agda.tex [hunk 1]
```diff
@@ -783,41 +783,30 @@ \section{Homomorphism In Agda}
...
-properties of both a direct product and a direct sum. In many cases, the
-bi-product coincides with the direct product when certain conditions are met.
-However, bi-products and direct products may have distinct properties and
-behaviors for many algebraic structures. For the scope of the thesis, we do not
-consider this distinction. There is currently an
+Recall the definition of product, direct product, and direct sum discussed in
+Chapter 2. A co-product of an algebra is the dual of the product algebra
+\cite{sannella2012foundations}. A bi-product of algebra is an algebra that is
+both a product and a co-product. The standard library defines objects that are
+products of appropriate algebras and addresses them as direct products. In many
```
Comment (JacquesCarette) [inline Agda.tex]: addresses -> calls

File: references.bib [hunk 2]
```diff
@@ -678,4 +678,18 @@ @book{judson2020abstract
...
+  }
+  @misc{directproduct,
+    title = {Direct product and Direct Sum},
+    author = {R.A Bailey},
+    publisher = {Queen Mary University of London},
+    pages = {4},
+    url = {https://webspace.maths.qmul.ac.uk/r.a.bailey/MAS305/algnotes14.pdf},
+    year = {2023}
+  }
+  @misc{szabo2000linear,
```
Comment (JacquesCarette) [inline references.bib]: Is this truly a 'misc' or is there a better way to refer to this work?

**Review by JacquesCarette — CHANGES_REQUESTED** [review]

File: Agda.tex (hunk 1 above)
Comment (Akshobhya1234) [reply Agda.tex]: fixed

File: references.bib (hunk 2 above)
Comment (Akshobhya1234) [reply references.bib]: It is a book. I fixed it.

**Review by JacquesCarette — APPROVED** [review]

**Pull request merged**